tests/integ/test_cte.py

```diff
@@ -484,6 +478,7 @@ def test_join_table_function(session):
     assert count_number_of_ctes(df_result.queries["queries"][-1]) == 1
 
 
+@pytest.mark.skipif(IS_IN_STORED_PROC, reason="Cannot create temp table in stored proc")
```

Comment:
do you think we can use this function: https://github.com/snowflakedb/snowpark-python/blob/main/tests/utils.py#L147-L148
to create table with random name in stored proc?

Reply:
giving a random name is not an issue here. We cannot create team tables using CTAS in stored procedures. We have to do it using the two queries: 
1. create scoped temp table
2. insert into table

Reply:
thanks for the context! I'm not aware of the limitation of CTAS before

Reply:
do you think we can write this? or rather we want to test this CTAS behavior.
I think rewriting should be straightforward and it helps improve test coverage.
```
table_name = Utils.random_table_name()
session.create_dataframe(
[<data>...])
.save_as_table(table_name)